Algorithms/Search/FastSearcher.cs

```diff
@@ -21,18 +21,21 @@ public class FastSearcher
         /// <param name="item">Term to check against.</param>
         /// <returns>Index of first item that satisfies term.</returns>
         /// <exception cref="ItemNotFoundException"> Gets thrown when the given item couldn't be found in the array.</exception>
-        public int FindIndex(int[] array, int item) => FindIndex(array, item, 0);
-
-        private int FindIndex(int[] array, int item, int offset)
+        public int FindIndex(Span<int> array, int item)
```

Comment:
Codacy found an issue: [The Cyclomatic Complexity of this method is 12 which is greater than 10 authorized.](https://app.codacy.com/manual/siriak/C-Sharp/pullRequest?prid=6092683)